### PR TITLE
Refactor and fix more bugs in mute-project and vol-slider

### PR DESCRIPTION
**Pull request by GarboMuffin**
This fixes for the same bugs as https://github.com/ScratchAddons/ScratchAddons/pull/5339 and thus supersedes it:

 - Fixes https://github.com/ScratchAddons/ScratchAddons/issues/5282
 - Uses default volume when mute-project is disabled instead of always 100%
 - Added user-select: none to mute-project icon
 - Adds non-prefixed appearance: none

Although the way these bugs are fixed is not always the same.

And also includes these:

 - When unmuting, it now uses the volume before the project was muted instead of the default volume setting
 - The vol-slider muted indicator will still be visible in small stage mode
 - Various changes to make the addons work in Scratch mods without further patches

### CODE REVIEW

File: addons/vol-slider/module.js [hunk 1]
```diff
@@ -1,40 +1,43 @@
...
+
+export const onVolumeChanged = (callback) => {
+  callbacks.push(callback);
 };
 
 export const setup = (_vm) => {
   if (vm) return;
   vm = _vm;
-  vmVol = vm.runtime.audioEngine.inputNode.gain;
+  gainNode = vm.runtime.audioEngine.inputNode.gain;
```
Comment (Samq64) [inline addons/vol-slider/module.js:42]: > Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'inputNode')

I'm not sure why this is happening now, but not before.

Comment (GarboMuffin) [reply addons/vol-slider/module.js]: was my bug introduced by this PR, should be fixed now

**Comment by WorldLanguages** [comment]
In the future, we could add a volume indicator SVG that has 2 waves instead of restricting it to 1 and 3, because the icons are being moved to the extension anyway

**Review by Samq64 — APPROVED** [review]
Everything works as expected and all the bug fixes mentioned work. You could also add yourself to the addon credits if you want.

**Pull request merged**